### PR TITLE
ar71xx: add support for TL-WA801ND-V4

### PR DESCRIPTION
**Pull request by JulianMa**
Addes Support for the TL-WA801ND-V4 by TP-Link.
It's almost the same hardware as in v3, just the GPIO mappings
for the led's and Reset/WPS Buttons changed.
There is no known issue.

Signed-off-by: Julian Maczuga <julian.maczuga@udo.edu>

I have running this build on my device for almost a year now and i havn't found any issues yet.

### CODE REVIEW

File: target/linux/ar71xx/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -794,6 +794,11 @@ tl-wa801nd-v3)
 	ucidef_set_led_netdev "lan" "LAN" "tp-link:green:lan" "eth1"
 	ucidef_set_led_wlan "wlan" "WLAN" "tp-link:green:wlan" "phy0tpt"
 	;;
+tl-wa801nd-v4)
+	ucidef_set_led_wlan "wlan" "WLAN" "tp-link:green:wlan" "phy0tpt"
+	ucidef_set_led_netdev "lan" "LAN" "tp-link:green:lan" "eth1"
+	ucidef_set_led_power "power" "POWER GREEN" "tp-link:green:power" "1"
```
Comment (adrianschmutzler) [inline target/linux/ar71xx/base-files/etc/board.d/01_leds]: Could be merged with v3 and condition for power part.

File: target/linux/ar71xx/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -260,6 +260,9 @@ ar71xx_setup_interfaces()
 	tl-wa801nd-v3)
 		ucidef_set_interface_lan "eth1"
 		;;
+	tl-wa801nd-v4)
```
Comment (adrianschmutzler) [inline target/linux/ar71xx/base-files/etc/board.d/02_network]: Merge with v3?

File: target/linux/ar71xx/config-4.9 [hunk 3]
```diff
@@ -198,6 +198,7 @@ CONFIG_ATH79=y
 # CONFIG_ATH79_MACH_TL_WA701ND_V2 is not set
 # CONFIG_ATH79_MACH_TL_WA7210N_V2 is not set
 # CONFIG_ATH79_MACH_TL_WA801ND_V3 is not set
+CONFIG_ATH79_MACH_TL_WA801ND_V4=y
```
Comment (obsy) [inline target/linux/ar71xx/config-4.9]: Use target/linux/ar71xx/tiny/config-default for 4M device.

File: target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wa801nd-v4.c [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	ath79_eth0_data.speed = SPEED_100;
+	ath79_eth0_data.phy_mask = BIT(4);
+	ath79_init_mac(ath79_eth0_data.mac_addr, mac, 1);
+	ath79_register_eth(0);
+
+	ath79_init_mac(tmpmac, mac, 0);
+	ath79_register_wmac(ee, tmpmac);
+}
+
+static void __init tl_wa801n_v4_setup(void)
```
Comment (juppin) [inline target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wa801nd-v4.c]: This function could also be merged into file mach-tl-wa801nd-v3.c

File: target/linux/ar71xx/base-files/etc/board.d/01_leds (hunk 1 above)
Comment (r3sist3nt) [reply target/linux/ar71xx/base-files/etc/board.d/01_leds]: You'r right, i will merge this.

File: target/linux/ar71xx/base-files/etc/board.d/02_network (hunk 2 above)
Comment (r3sist3nt) [reply target/linux/ar71xx/base-files/etc/board.d/02_network]: I will merge this, as mentioned in the comment above.

File: target/linux/ar71xx/config-4.9 (hunk 3 above)
Comment (r3sist3nt) [reply target/linux/ar71xx/config-4.9]: You are right. Will change this.

File: target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wa801nd-v4.c (hunk 4 above)
Comment (juppin) [reply target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wa801nd-v4.c]: duplicate code is duplicate code...

Look at this example https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wr841n-v9.c
There are 5 devices with different button and led configs merged into one mach file...

Comment (r3sist3nt) [reply target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wa801nd-v4.c]: Alright, please forgive me my mistakes, iam pretty new to openwrt dev.
I will merge the files.

**Comment by blogic** [comment]
no update in over 2 months, please reopen if still relevant.

**Pull request closed without merging**